lib/Target/Lattigo/LattigoEmitter.cpp

```diff
@@ -224,9 +225,16 @@ LogicalResult LattigoEmitter::printOperation(arith::ConstantOp op) {
           })
           .Case<DenseElementsAttr>([&](DenseElementsAttr denseAttr) {
             if (succeeded(denseAttr.tryGetValues<APInt>())) {
-              valueString = "[]int64{";
-              for (auto value : denseAttr.getValues<APInt>()) {
-                valueString += std::to_string(value.getSExtValue()) + ", ";
+              if (denseAttr.getType().getElementType().isInteger(1)) {
```

Comment:
nit: i was noticing the i1 / bool case isn't handled for the IntegerAttr case above, but then in convertType it is handlded below.

Maybe you can remove this typeswitch and instead rely on `auto res = convertType(valueAttr.getType())`?

Reply:
Now `arith.constant` is _honest_ on type: what is in IR is converted to corresponding type using convertType. The reason int64/float64 is delibrately here was that Encode/Decode only accepts `int64[]/float64[]` and we have the following common pattern

```
cst = arith.constant <4xi16>
Decode(pt, cst)
```

so for convenience I directly convert i16 to i64. Now the extra step is done in DecodeOp.